src/transform-layer.coffee

```diff
@@ -32,3 +38,81 @@ class TransformLayer extends Layer
     @pendingChangeOldExtent = null
 
     @emitter.emit "did-change", {position: startPosition, oldExtent, newExtent}
+
+  toSourcePosition: (position, clip) ->
+    iterator = @buildIterator()
+    iterator.seek(position, clip)
+    iterator.getSourcePosition()
+
+  fromSourcePosition: (sourcePosition, clip) ->
+    iterator = @buildIterator()
+    iterator.seekToSourcePosition(sourcePosition, clip)
+    iterator.getPosition()
+
+class TransformLayerIterator
+  clipping: undefined
+
+  constructor: (@layer, sourceIterator) ->
+    @position = Point.zero()
+    @sourcePosition = Point.zero()
+    @transformBuffer = new OperatorIterator(@layer.operator, sourceIterator)
+
+  next: ->
+    if next = @transformBuffer.next()
+      {@position, @sourcePosition, content, @clipping} = next
+      {value: content, done: false}
+    else
+      {value: undefined, done: true}
+
+  seek: (position, clip=CLIP_BACKWARD) ->
+    @position = Point.zero()
+    @sourcePosition = Point.zero()
+    @transformBuffer.reset(@position, @sourcePosition)
+    return if position.isZero()
+
+    until @position.compare(position) >= 0
+      lastPosition = @position
+      lastSourcePosition = @sourcePosition
+      {done} = @next()
+      return if done
+
+    if @clipping? and @position.compare(position) > 0
+      switch clip
+        when CLIP_FORWARD
+          return
+        when CLIP_BACKWARD
+          @sourcePosition = lastSourcePosition
+          return
+
+    unless @position.compare(position) is 0
+      overshoot = position.column - lastPosition.column
+      lastSourcePosition.column += overshoot
+      @position = position
+      @sourcePosition = lastSourcePosition
+    @transformBuffer.reset(@position, @sourcePosition)
+
+  seekToSourcePosition: (position) ->
+    @position = Point.zero()
+    @sourcePosition = Point.zero()
+    @transformBuffer.reset(@position, @sourcePosition)
+    return if position.isZero()
+
+    until @sourcePosition.compare(position) >= 0
+      lastPosition = @position
+      lastSourcePosition = @sourcePosition
+      {done} = @next()
+      break if done
+
+    return if @clipping?
+
+    overshoot = position.column - lastSourcePosition.column
```

Comment:
Related to why 0acfae82ae83cdb727dfe9235e658f9d0f1a6fc3 is failing, I think that the overshoot calculation is the culprit. I feel like we should take into account both `x` and `y` when calculating it and to clip to the next row where appropriate.

How do you think we could proceed here?